api/http/proxy/manager.go

```diff
@@ -51,6 +51,9 @@ func (manager *Manager) createDockerProxy(endpointURL *url.URL, tlsConfig *porta
 		}
 		return manager.proxyFactory.newDockerHTTPProxy(endpointURL, false), nil
 	}
+	if endpointURL.Scheme == "npipe" {
```

Comment:
I think that we can assume `npipe://` or `unix://` here depending on the platform, no need for that check.

Simply use:

```go
return manager.proxyFactory.newLocalProxy(endpointURL.Path), nil
```

See my comment above regarding the `local_unix.go` and `local_windows.go` files.